common/common.go

```diff
@@ -0,0 +1,221 @@
+package common
+
+import (
+	"flag"
+	"fmt"
+	"github.com/logzio/logzio_terraform_client/grafana_alerts"
+	"github.com/prometheus/prometheus/model/rulefmt"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+	"k8s.io/klog/v2"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strconv"
+	"time"
+)
+
+const (
+	LetterBytes   = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+	letterIdxBits = 6                    // 6 bits to represent a letter index
+	letterIdxMask = 1<<letterIdxBits - 1 // All 1-bits, as many as letterIdxBits
+	letterIdxMax  = 63 / letterIdxBits
+	TypeSlack     = "slack"
+	TypeEmail     = "email"
+	TypePagerDuty = "pagerduty" // # of letter indices fitting in 63 bits
+)
+
+// NewConfig creates a Config struct, populating it with values from command-line flags and environment variables.
+func NewConfig() *Config {
+	// Define flags
+	helpFlag := flag.Bool("help", false, "Display help")
+	rulesConfigmapAnnotation := flag.String("rules-annotation", "prometheus.io/kube-rules", "Annotation that states that this configmap contains prometheus rules")
+	alertManagerConfigmapAnnotation := flag.String("alertmanager-annotation", "prometheus.io/kube-alertmanager", "Annotation that states that this configmap contains alertmanager configuration")
+	logzioAPITokenFlag := flag.String("logzio-api-token", "", "LOGZIO API token")
+	logzioAPIURLFlag := flag.String("logzio-api-url", "https://api.logz.io", "LOGZIO API URL")
+	rulesDSFlag := flag.String("rules-ds", "", "name of the data source for the alert rules")
+	envIDFlag := flag.String("env-id", "my-env", "environment identifier, usually cluster name")
+	workerCountFlag := flag.Int("workers", 2, "The number of workers to process the alerts")
+	ignoreSlackTextFlag := flag.Bool("ignore-slack-text", false, "Ignore slack text field")
+	ignoreSlackTitleFlag := flag.Bool("ignore-slack-title", false, "Ignore slack title field")
+
+	// Parse the flags
+	flag.Parse()
+
+	if *helpFlag {
+		flag.PrintDefaults()
+		os.Exit(0)
+	}
+
+	// Environment variables have lower precedence than flags
```

Comment:
From what I understand, when you use "getEnvWithFallback" you first check if the env var exists, and if not - using the value from the flag. Doesn't that mean that env vars have priority over flags?

Reply:
Yes env var has a priority over flags in this program